pkg/config/files.go

```diff
@@ -80,31 +80,37 @@ func parse(contents []byte) (*Config, error) {
 	if err := yaml.Unmarshal(contents, c); err != nil {
 		return nil, fmt.Errorf("Unable to decode configuration: %v", err)
 	}
-	if err := c.fillDefaults(); err != nil {
-		return nil, fmt.Errorf("Invalid configuration: %v", err)
+	return c, nil
+}
+
+func getActiveConfigFromYAML(contents []byte) (*Config, error) {
+	userSettings, err := parse(contents)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing config file %q: %v", configFile, err)
 	}
-	if err := c.updateComputedValues(); err != nil {
+
+	results := &Config{}
+	results.userSettings = userSettings
+	results.incorporateUserSettings()
```

Comment:
Shall we invoke `incorporateUserSettings` after `fillDefaults`? So that user settings can always overwrite the defaults.

Reply:
Yes, great point. The fillDefaults function tries to be smart, but it might be better to have that be simple and move the smarts to the other function. I will try to do that tomorrow.

Reply:
Done